### DISCUSSION TITLE
Health check exceeded timeout - how to debug?

### DISCUSSION THREAD

**Discussion by smr99** [nextcloud/all-in-one#2126] [Questions] [answered]
I've installed the AIO docker images, got nextcloud up and running, connected from phone apps, etc.  All seemed to work for a couple days.  Last Sunday, I realised that things weren't working.  "docker ps" showed four of the containers as healthy but "nextcloud/aio-apache" was "unhealthy".  I didn't really know what to do, but after stopping all containers, removing the master container, then re-starting it I was able to get the initial port-8088 interface where it showed there were updates to do.  I clicked on that and most worked, but apache failed and showed a red (stopped) indicator.  I repeated the master container stop/remove/re-start and the system came back up clean.  All seemed well - logged in using web, apps, etc.

Today (four days later), I find apache is unhealthy again.  On another question I saw someone used "docker inspect nextcloud-aio-apache" to check things.  I did and noticed that it reports "Health check exceeded timeout (30s)" with a failing streak of almost 5000.  I suppose this is my issue, but I don't know how to debug it.  Any tips?

**Comment by szaimen** [comment, marked as answer]
Hi, here is the health check script that is used. Hopefully it helps you figuring things out: https://github.com/nextcloud/all-in-one/blob/main/Containers/apache/healthcheck.sh And indeed does the healthcheck work here as expected so it is most likely caused by your network infrastructure.

**Comment by thamow** [comment]
Hi @smr99, I seem to have the same issue. Do you have any info for me about how you fixed it (if you managed to that is)? Do you have Nextcloud installed behind a reverse proxy? Because I do and I'm not quite sure whether that causes the problem...

**Reply by smr99** [reply]
I kind of ignored things (Nextcloud isn't critical to me in any way) and they seemed to get better at some point.  The containers have remained healthy for several months now.  I assumed that the health script got fixed, but never looked into it.

I don't think I have a reverse proxy -- at least not one that I configured myself.

**Reply by thamow** [reply]
Alright, thank you nonetheless!
